libs/ha/ha_common_libs_k8s.py

```diff
@@ -1845,3 +1846,116 @@ def dnld_obj_verify_chcksm(self, s3_test_obj, bucket, obj, download_path, upload
             return False, download_checksum
         LOGGER.info("Successfully downloaded the object and verified the checksum")
         return True, download_checksum
+
+    @staticmethod
+    def parallel_put_object(event, s3_test_obj, bkt_name: str, obj_name: str, output=None,
+                            **kwargs):
+        """
+        Function to upload object
+        Upload an object to a versioning enabled/suspended bucket and return list of dictionary of
+        uploaded versions in parallel or non parallel.
+        :param event: event to intimate thread about main thread operations
+        :param s3_test_obj: Object of the s3 test lib
+        :param bkt_name: Bucket name for calling PUT Object
+        :param obj_name: Object name for calling PUT Object
+        :param output: Output queue in which results should be put
+        :keyword file_path: File path that can be used for PUT Object call
+        :keyword count: Count for number of PUT object call
+        :keyword chk_null_version: True, if 'null' version id is expected, else False
+        :keyword background: Set to true if background function call
+        :keyword is_unversioned: Set to true if object is uploaded to an unversioned bucket
+        Can be used for setting up pre-existing objects before enabling/suspending bucket
+        versioning
+        :return: Tuple (bool, list)
+        """
+        chk_null_version = kwargs.get("chk_null_version", False)
+        file_path = kwargs.get("file_path")
```

Comment:
can we have some default value for this as well?

Reply:
> can we have some default value for this as well?

This is Mandatory for put operation as we want to put same object every time